File: FitstBack/src/main/java/ma/ensaj/hotel/entity/Chambre.java
package ma.ensaj.hotel.entity;

import jakarta.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Chambre {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String type;
    private Double prix;
    private Boolean disponible;
    public Chambre() {
    }

    public Chambre(Long id, String type, Double prix, Boolean disponible) {
        this.id = id;
        this.type = type;
        this.prix = prix;
        this.disponible = disponible;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Double getPrix() {
        return prix;
    }

    public void setPrix(Double prix) {
        this.prix = prix;
    }

    public Boolean getDisponible() {
        return disponible;
    }

    public void setDisponible(Boolean disponible) {
        this.disponible = disponible;
    }
}

File: FrontEnd Android/app/src/main/java/abdellah/project/retrofite_rest/entity/Chambre.kt
package abdellah.project.retrofite_rest.entity
class Chambre {

    private var id: Long? = null
    private var type: String? = null
    private var prix: Double? = null
    private var disponible: Boolean? = null

    // Constructeur principal
    constructor()

    constructor(id: Long?, type: String?, prix: Double?, disponible: Boolean?) {
        this.id = id
        this.type = type
        this.prix = prix
        this.disponible = disponible
    }

    // Getter et Setter pour id
    fun getId(): Long? {
        return id
    }

    fun setId(id: Long?) {
        this.id = id
    }

    // Getter et Setter pour type
    fun getType(): String? {
        return type
    }

    fun setType(type: String?) {
        this.type = type
    }

    // Getter et Setter pour prix
    fun getPrix(): Double? {
        return prix
    }

    fun setPrix(prix: Double?) {
        this.prix = prix
    }

    // Getter et Setter pour disponible
    fun getDisponible(): Boolean? {
        return disponible
    }

    fun setDisponible(disponible: Boolean?) {
        this.disponible = disponible
    }

    override fun toString(): String {
        return "Chambre(id=$id, type=$type, prix=$prix, disponible=$disponible)"
    }
}
